src/components/MultiGestureCanvas/usePinchGesture.ts

```diff
@@ -127,58 +127,58 @@ const usePinchGesture = ({
                 return;
             }
 
-            const newZoomScale = pinchScale.value * evt.scale;
-
+            const newZoomScale = pinchScale.get() * evt.scale;
+            const zoomScaleValue = zoomScale.get();
             // Limit the zoom scale to zoom range including bounce range
-            if (zoomScale.value >= zoomRange.min * ZOOM_RANGE_BOUNCE_FACTORS.min && zoomScale.value <= zoomRange.max * ZOOM_RANGE_BOUNCE_FACTORS.max) {
-                zoomScale.value = newZoomScale;
-                currentPinchScale.value = evt.scale;
+            if (zoomScaleValue >= zoomRange.min * ZOOM_RANGE_BOUNCE_FACTORS.min && zoomScaleValue <= zoomRange.max * ZOOM_RANGE_BOUNCE_FACTORS.max) {
+                zoomScale.set(newZoomScale);
+                currentPinchScale.set(evt.scale);
 
                 triggerScaleChangedEvent();
             }
 
             // Calculate new pinch translation
             const adjustedFocal = getAdjustedFocal(evt.focalX, evt.focalY);
-            const newPinchTranslateX = adjustedFocal.x + currentPinchScale.value * pinchOrigin.x.value * -1;
-            const newPinchTranslateY = adjustedFocal.y + currentPinchScale.value * pinchOrigin.y.value * -1;
+            const newPinchTranslateX = adjustedFocal.x + currentPinchScale.get() * pinchOrigin.x.get() * -1;
+            const newPinchTranslateY = adjustedFocal.y + currentPinchScale.get() * pinchOrigin.y.get() * -1;
 
             // If the zoom scale is within the zoom range, we perform the regular pinch translation
             // Otherwise it means that we are "overzoomed" or "underzoomed", so we need to bounce back
-            if (zoomScale.value >= zoomRange.min && zoomScale.value <= zoomRange.max) {
-                pinchTranslateX.value = newPinchTranslateX;
-                pinchTranslateY.value = newPinchTranslateY;
+            if (zoomScaleValue >= zoomRange.min && zoomScaleValue <= zoomRange.max) {
```

Comment:
I believe `zoomScaleValue` is outdated since we might have modified it in `zoomScale.set(newZoomScale);` above.